openedx/core/lib/edx_api_utils.py

```diff
@@ -45,8 +47,9 @@ def get_edx_api_data(api_config, user, resource, resource_id=None, querystring=N
             return cached
 
     try:
-        jwt = get_id_token(user, api_config.OAUTH2_CLIENT_NAME)
-        api = EdxRestApiClient(api_config.internal_api_url, jwt=jwt)
+        if not api:
```

Comment:
This is what I was looking for. However, it's missing test coverage. An easy way to verify the behavior would be to mock `EdxRestApiClient` and verify that it's not called when an instantiated client is provided.